docs/entity-definition.md

```diff
@@ -1,29 +1,106 @@
  
 So what do we mean by remote entities? A remote entity is an entity, as understood above, that 1) is not the local host where either the agent and integration live or 2) is not equivalent to a “host”. In this way, we offer support to monitor a whole host of things, forgive the bad pun, that we might not otherwise be able to, due to being able to get an agent on the host or the architecture itself requiring the concept of multiple “entities”.
  
-For Example:
- 
-An engineer installs the infrastructure agent on host1 and configures the out-of-the-box mysql integration to monitor a MySQL server running on host2. Host1 would be an entity as far as the Infrastructure entity is concerned and the MySQL servers on Host2 would be a “remote entity”.
- 
-An engineer installs the infrastructure agent on host1 and configures the out-of-the-box kubernetes integration and collects metrics and inventory about the whole cluster, replica sets, pods and nodes.
+> For Example:
+>
+> An engineer installs the infrastructure agent on host1 and configures the out-of-the-box mysql integration to monitor a MySQL server running on host2. Host1 would be an entity as far as the Infrastructure entity is concerned and the MySQL servers on Host2 would be a “remote entity”.
+>
+> An engineer installs the infrastructure agent on host1 and configures the out-of-the-box kubernetes integration and collects metrics and inventory about the whole cluster, replica sets, pods and nodes.
+
+
+## Entity uniqueness
+
+Entity uniqueness is provided by its **key**.
+
+An entity `key` is formed by its `namespace` (or *type*) and `name`. These fields are concatenated with `:` separator.
+
+> For instance: `integration.Entity("name", "ns")` would create an entity with a `key` value `ns:name`.   
+
+Entities are attached to a user account. Each integration is responsible for providing uniqueness for its entities.
+
+It's up to the integration developer to define an entity `name` that's **unique** to its user account. An entity name should map uniquely to the entity being monitored.
+
+Imagine a situation where you are monitoring two Kafka clusters, one meant for development and the other for production, where you have a topic with the same name on each. The entity needs to map uniquely to the development topic and the production topic so that there is no collision.
+
+
+## Entity naming guidelines
+
+An entity name is an string identifier by which user uniquely identifies an entity.
+
+Entity names should:
+
+- provide enough information to be uniquely identified
+- easily identifiable for a user to locate what they refer to
+
+**Key point**: bundle as many identifier-attributes as required but not more.
+
+Usually an *endpoint* is a good starting point to identify a service.
+
+> Eg: `host:port`
+
+
+### Entity name composition
+
+Names could be composed of several fields, in this case we call these **identifier-attributes**.
 
-### Metadata decoration
+#### Identifier attributes
```

Comment:
maybe it's quite obvious, but should we say the identifier attribute key should not use any special character like `:` or `=`?

Reply:
good point

Reply:
pefecgt